File: hibernate-examples/src/test/java/org/hibernate/examples/jpa/config/JpaAccountRepository.java
package org.hibernate.examples.jpa.config;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.jpa.repository.QueryHints;
import org.springframework.data.repository.query.Param;

import javax.persistence.QueryHint;

/**
 * org.hibernate.examples.jpa.config.JpaAccountRepository
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2013. 11. 28. 오후 10:34
 */
public interface JpaAccountRepository extends JpaRepository<JpaAccount, Long> {

  @Query(value = "select a from JpaAccount a where a.name=:name")
  @QueryHints({@QueryHint(name = "org.hibernate.cacheable", value = "true")})
  JpaAccount findByName(@Param("name") final String name);
}

File: hibernate-scala-examples/src/test/scala/org/hibernate/examples/tests/jpa/config/JpaAccountRepository.scala
package org.hibernate.examples.tests.jpa.config

import org.springframework.data.jpa.repository.{Query, JpaRepository}
import org.springframework.data.repository.query.Param

/**
 * Spring Data JPA 용 Repository 입니다.
 * NOTE: QueryDSL for Scala 사용법을 몰라 QueryDslPredicateExecutor는 사용하지 못합니다.
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2014. 1. 9. 오후 4:28
 */
trait JpaAccountRepository extends JpaRepository[JpaAccount, java.lang.Long] {
    // with QueryDslPredicateExecutor[JpaAccount] {

    @Query(value = "select a from JpaAccount a where a.name = :name")
    def findByName(@Param("name") name: String): JpaAccount

}
